satpy/composites/__init__.py

```diff
@@ -361,11 +369,16 @@ def __call__(self, projectables, **kwargs):
         if len(projectables) != 1:
             raise ValueError("Can't have more than one dataset for a categorical data composite")
 
-        data = projectables[0].astype(int)
-        res = data.data.map_blocks(self._getitem, self.lut, dtype=self.lut.dtype)
+        data = projectables[0]
+        if not np.issubdtype(data.dtype, np.integer):
+            raise TypeError(f"{type(self).__name__:s} can only be used on "
```

Comment:
```suggestion
            raise TypeError(f"{self.__class__.__name__:s} can only be used on "
```

I think this works and is more clear to me.